proof.go

```diff
@@ -465,6 +465,203 @@ func (proof Proof) VerifyInclusion(h hash.Hash, nid namespace.ID, leavesWithoutN
 	return res
 }
 
+// VerifySubtreeRootInclusion verifies that a set of subtree roots is included in
+// an NMT.
+// Note: This method is Celestia specific.
+// It makes the following assumptions:
+// - The subtree roots are created according to the ADR-013
+// https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md
+// - The tree's number of leaves is a power of two
+// Using this method without making sure the above assumptions are respected
+// can return invalid results.
+// The subtreeWidth is also defined in ADR-013.
```

Comment:
That was my worry too at the beginning, but then I checked the feasibility of extracting this to a separate repo, like in node where it will be used, or even put it in a separate package, but found that we will need to export a lot of methods in nmt not to end up copy-pasting the code.

So, I decided to keep it here... I mention extensively in the docs that this is Celestia specific, so I think it should be fine. I could even change the function names to mention Celestia if that works for you.

Otherwise, I am open to any suggestion.

Reply:
I think we should open an issue to track this. In celestia-app maybe.

Reply:
https://github.com/celestiaorg/celestia-app/issues/3615